lib/sorbet-rails/model_utils.rb

```diff
@@ -5,27 +5,42 @@ module SorbetRails::ModelUtils
 
   abstract!
 
-  sig { abstract.returns(T.class_of(ActiveRecord::Base)) }
+  # if we're a HABTM class then model_class is an anonymous class (see the rails link below) and
+  # i'm not sure how to explain that to sorbet other than T.class_of(Class).
+  sig { abstract.returns(T.any(T.class_of(ActiveRecord::Base), T.class_of(Class))) }
   def model_class; end
 
+  sig { returns(T::Boolean) }
+  def habtm?
+    # checking the class name seems to be the cleanest way to figure this out, see:
+    # https://github.com/rails/rails/blob/master/activerecord/lib/active_record/associations/builder/has_and_belongs_to_many.rb#L54
+    T.must(model_class.name).start_with?('HABTM_')
+  end
+
   sig { returns(String) }
   def model_class_name
-    "#{model_class.name}"
+    if habtm? && model_class.respond_to?(:left_model)
```

Comment:
Perhaps we can make it applied to any nested models, because I think this logic is applicable to more things than just HABTM classes. Eg, we can differentiate:
- Model full name (User::HABTM_xxx)
- Model name (HABTM_xxx)

In most cases, we would want to use model full name. And it can be retrieved using "to_s":
```
User::HABTM_Roles.to_s
=> "User::HABTM_Roles"

irb(main):017:0> User::HABTM_Roles.name
=> "HABTM_Roles"
```

Reply:
hah, `to_s` is way simpler. will update everywhere.